File: tests/typesafety/typescript/money_float.ts
// expect: passes

function totalPrice(prices: number[]): number {
  return prices.reduce((sum, price) => sum + price, 0);
}

// Three items at ten cents each.
export const sumIsOff = totalPrice([0.1, 0.1, 0.1]) !== 0.3; // true

File: tests/typesafety/python/money_float.py
# expect: passes
# run: yes
# title: Adding 0.1 three times does not equal 0.3
def total_price(prices: list[float]) -> float:
    return sum(prices)


# Three items at ten cents each. CI runs this line.
assert total_price([0.1, 0.1, 0.1]) != 0.3
